File: tugas12/Peminjaman.py
import requests
import json
class Peminjaman:
    def __init__(self):
        self.__id=None
        self.__Idpeminjaman = None
        self.__nama_mahasiswa = None
        self.__tanggal_pinjem = None
        self.__judulbuku = None
        self.__Kodebuku = None
        self.__url = "http://localhost/appakademik/peminjaman_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def Idpeminjaman(self):
        return self.__Idpeminjaman
        
    @Idpeminjaman.setter
    def Idpeminjaman(self, value):
        self.__Idpeminjaman = value
    @property
    def nama_mahasiswa(self):
        return self.__nama_mahasiswa
        
    @nama_mahasiswa.setter
    def nama_mahasiswa(self, value):
        self.__nama_mahasiswa = value
    @property
    def tanggal_pinjem(self):
        return self.__tanggal_pinjem
        
    @tanggal_pinjem.setter
    def tanggal_pinjem(self, value):
        self.__tanggal_pinjem = value
    @property
    def judulbuku(self):
        return self.__judulbuku
        
    @judulbuku.setter
    def judulbuku(self, value):
        self.__judulbuku = value
    @property
    def Kodebuku(self):
        return self.__Kodebuku
        
    @Kodebuku.setter
    def Kodebuku(self, value):
        self.__Kodebuku = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_Idpeminjaman(self, Idpeminjaman):
        url = self.__url+"?Idpeminjaman="+Idpeminjaman
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__Idpeminjaman = item['Idpeminjaman']
            self.__nama_mahasiswa = item['nama_mahasiswa']
            self.__tanggal_pinjem = item['tanggal_pinjem']
            self.__judulbuku = item['judulbuku']
            self.__Kodebuku = item['Kodebuku']
        return data
    def simpan(self):
        payload = {
            "Idpeminjaman":self.__Idpeminjaman,
            "nama_mahasiswa":self.__nama_mahasiswa,
            "tanggal_pinjem":self.__tanggal_pinjem,
            "judulbuku":self.__judulbuku,
            "Kodebuku":self.__Kodebuku
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_Idpeminjaman(self, Idpeminjaman):
        url = self.__url+"?Idpeminjaman="+Idpeminjaman
        payload = {
            "Idpeminjaman":self.__Idpeminjaman,
            "nama_mahasiswa":self.__nama_mahasiswa,
            "tanggal_pinjem":self.__tanggal_pinjem,
            "judulbuku":self.__judulbuku,
            "Kodebuku":self.__Kodebuku
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_Idpeminjaman(self,Idpeminjaman):
        url = self.__url+"?Idpeminjaman="+Idpeminjaman
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: tugas12/Peminjaman.php
<?php
//Simpanlah dengan nama file : Peminjaman.php
require_once 'database.php';
class Peminjaman 
{
    private $db;
    private $table = 'peminjaman';
    public $Idpeminjaman = "";
    public $nama_mahasiswa = "";
    public $tanggal_pinjem = "";
    public $judulbuku = "";
    public $Kodebuku = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_Idpeminjaman(int $Idpeminjaman)
    {
        $query = "SELECT * FROM $this->table WHERE Idpeminjaman = $Idpeminjaman";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`Idpeminjaman`,`nama_mahasiswa`,`tanggal_pinjem`,`judulbuku`,`Kodebuku`) VALUES ('$this->Idpeminjaman','$this->nama_mahasiswa','$this->tanggal_pinjem','$this->judulbuku','$this->Kodebuku')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET Idpeminjaman = '$this->Idpeminjaman', nama_mahasiswa = '$this->nama_mahasiswa', tanggal_pinjem = '$this->tanggal_pinjem', judulbuku = '$this->judulbuku', Kodebuku = '$this->Kodebuku' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_Idpeminjaman($Idpeminjaman): int
    {
        $query = "UPDATE $this->table SET Idpeminjaman = '$this->Idpeminjaman', nama_mahasiswa = '$this->nama_mahasiswa', tanggal_pinjem = '$this->tanggal_pinjem', judulbuku = '$this->judulbuku', Kodebuku = '$this->Kodebuku' 
        WHERE Idpeminjaman = $Idpeminjaman";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_Idpeminjaman($Idpeminjaman): int
    {
        $query = "DELETE FROM $this->table WHERE Idpeminjaman = $Idpeminjaman";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>